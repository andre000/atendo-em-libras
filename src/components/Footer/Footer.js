import React, { PureComponent } from 'react'
import styled from "styled-components";

import { Container } from '../Container';
import logo from '../../assets/images/img_atendo_em_libras_logo_blue.svg';

class Footer extends PureComponent {
  render = () => {
    return (
      <div className={this.props.className}>
        <Container>
          <img src={logo} alt={'Logo projeto Atendo em Libras'}></img>
        </Container>
      </div>
    )
  }
}

const StyledFooter = styled(Footer)`
  width: 100%;
  height: 30vh;
  display: flex;
  flex-flow: row;
  align-items: center;
  justify-content: space-between;
  color: white;
  background-color: #FFFFFF;
`

export { StyledFooter as Footer }