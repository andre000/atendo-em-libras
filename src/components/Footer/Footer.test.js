import React from 'react'
import { Footer } from './Footer'
import { render } from '@testing-library/react'
import 'jest-styled-components'

describe('Footer component', () => {
  let view

  beforeEach(() => {
    view = render(<Footer />)
  })

  it('renders footer correctly', () => {
    expect(view.container).toMatchSnapshot()
  })

  it('should include Atendo em Libras email with label', () => {
    expect(view.getByText('DÃºvidas? Entre em contato em:')).toBeInTheDocument()
    expect(view.getByText('atendoemlibras@gmail.com')).toBeInTheDocument()
  })
})
