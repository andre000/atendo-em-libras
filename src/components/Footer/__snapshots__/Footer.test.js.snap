// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders footer correctly 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white;
  background-color: #ffffff;
  border-top: solid #0000001a 1px;
  -webkit-flex-flow: row;
  -ms-flex-flow: row;
  flex-flow: row;
  margin-top: 50px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row;
  -ms-flex-flow: row;
  flex-flow: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  height: 5em;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-flex-flow: row;
  -ms-flex-flow: row;
  flex-flow: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  height: 5em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 img {
  height: 1.8em;
}

.c5 p,
.c5 a {
  color: black;
  font-size: 1.6em;
  font-weight: 600;
}

.c5 *:last-child {
  font-weight: 400;
}

.c5:first-child {
  margin-right: 100px;
}

.c1 {
  max-width: 1500px;
  width: 100%;
  box-sizing: border-box;
  padding: 0 80px;
  margin: 0 auto;
}

@media (max-width:800px) {
  .c2 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
    margin: 58px 0 58px 0;
  }
}

@media (min-width:801px) {
  .c2 {
    height: 20vh;
  }
}

@media (max-width:800px) {
  .c4 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }
}

@media (max-width:800px) {
  .c5 {
    margin-top: 40px;
  }
}

@media (max-width:800px) {
  .c1 {
    padding: 0 10vw;
  }
}

<div>
  <footer
    class="c0"
  >
    <div
      class="c1"
    >
      <section
        class="c2"
      >
        <img
          alt="Logo Atendo em Libras"
          class="c3"
          src="img_atendo_em_libras_logo_blue.svg"
        />
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <p>
              Apoio:
            </p>
            <img
              alt="Logo ThoughtWorks"
              src="img_thoughtworks_logo_black.svg"
            />
          </div>
          <div
            class="c5"
          >
            <p>
              DÃºvidas? Entre em contato em:
            </p>
            <a
              href="mailto:atendoemlibras@gmail.com"
            >
              atendoemlibras@gmail.com
            </a>
          </div>
        </div>
      </section>
    </div>
  </footer>
</div>
`;
